Sources/TuistCoreTesting/Mappers/MockWorkspaceMapper.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+import TSCBasic
+import TuistGraph
+@testable import TuistCore
+
+public final class MockWorkspaceMapper: WorkspaceMapping {
+    public var mapStub: ((WorkspaceWithProjects) -> (WorkspaceWithProjects, [SideEffectDescriptor]))?
+    public func map(workspace: WorkspaceWithProjects) throws -> (WorkspaceWithProjects, [SideEffectDescriptor]) {
+        mapStub?(workspace) ?? (WorkspaceWithProjects(workspace: .test(), projects: [.test()]), [])
```

Comment:
might be a good idea to define a `test()` helper function for `WorkspaceWithProjects` to be consistent with the codebase

Reply:
Not using this anymore in this PR but adding it for the future users 👍🏼